File: Nov2020/17Nov/FunGame<Capillary>.java
/*
Problem Name: Fun Game <Capillary>
Problem Link: https://www.hackerearth.com/practice/data-structures/stacks/basics-of-stacks/practice-problems/algorithm/fun-game-91510e9f/
*/
import java.io.*;
import java.util.*;
public class FunGame<Capillary> {
    static int[] funGame(int[] arr){
        // Write your code here
        int n = arr.length, a = 0, b = n - 1;
        List<Integer> l =new ArrayList<Integer>();
        for(;a<n && b>=0;){
            if(arr[a] > arr[b]){
                l.add(1);
                b--;
            }
            else if(arr[a] < arr[b]){
                l.add(2);
                a++;
            }
            else{
                l.add(0);
                a++;
                b--;
            }
        }
        int ans[] = new int[l.size()];
        for(int i=0;i<l.size();i++)
            ans[i] = l.get(i);
        return ans;
    }
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter wr = new PrintWriter(System.out);
         int n = Integer.parseInt(br.readLine().trim());
         String[] arr_arr = br.readLine().split(" ");
         int[] arr = new int[n];
         for(int i_arr=0; i_arr<arr_arr.length; i_arr++)
         {
         	arr[i_arr] = Integer.parseInt(arr_arr[i_arr]);
         }

         int[] out_ = funGame(arr);
         System.out.print(out_[0]);
         for(int i_out_=1; i_out_<out_.length; i_out_++)
         {
         	System.out.print(" " + out_[i_out_]);
         }

         wr.close();
         br.close();
    }
}

File: Nov2020/17Nov/FunGame<Capillary>.cpp
/*
Problem Name: Fun Game <Capillary>
Problem Link: https://www.hackerearth.com/practice/data-structures/stacks/basics-of-stacks/practice-problems/algorithm/fun-game-91510e9f/
*/

#include<bits/stdc++.h>
#include<vector>
using namespace std;

vector<int> funGame (vector<int> arr) {
   // Write your code here
   vector<int>result;
   int a=0,b=arr.size()-1;
   
   while(a<arr.size() && b>=0){
      if(arr[a]>arr[b]){
         result.push_back(1);
         b--;
      }
      else if(arr[a]<arr[b]){
         result.push_back(2);
         a++;
      }
      else{
         result.push_back(0);
         a++;
         b--;
      }
   }
   return result;
}

int main() {

    ios::sync_with_stdio(0);
    cin.tie(0);
    int n;
    cin >> n;
    vector<int> arr(n);
    for(int i_arr=0; i_arr<n; i_arr++)
    {
    	cin >> arr[i_arr];
    }

    vector<int> out_;
    out_ = funGame(arr);
    cout << out_[0];
    for(int i_out_=1; i_out_<out_.size(); i_out_++)
    {
    	cout << " " << out_[i_out_];
    }
}
